crates/red_knot_python_semantic/src/semantic_index/symbol.rs

```diff
@@ -170,6 +174,13 @@ pub enum ScopeKind {
     Annotation,
     Class,
     Function,
+    Generator,
+}
+
+impl ScopeKind {
+    pub const fn is_generator(self) -> bool {
+        matches!(self, ScopeKind::Generator)
+    }
```

Comment:
> But we should add this scope kind to `ScopeId::is_function_like` above, since otherwise comprehension scopes just behave like function scopes.

Isn't that changed with [PEP 709 - Inlined comprehensions](https://peps.python.org/pep-0709/) in Python 3.12?

Reply:
> Isn't that changed with [PEP 709 - Inlined comprehensions](https://peps.python.org/pep-0709/) in Python 3.12?

The internal implementation in CPython no longer uses a function object for these scopes, following the changes Carl made to implement PEP 709. But the user-facing behaviour is the same in 99% of situations; semantically, they still behave like function scopes in almost every way

Reply:
I see, that makes sense. Thanks for clarifying.